python/ray/autoscaler/gcp/config.py

```diff
@@ -167,16 +152,15 @@ def construct_clients_from_provider_config(provider_config):
                 "formatted improperly.")
         credentials = service_account.Credentials.from_service_account_info(
             service_account_info)
-        return _create_crm_service_account(credentials), \
-            _create_iam_service_account(credentials), \
-            _create_compute_service_account(credentials)
+        return _create_crm(credentials), \
+            _create_iam(credentials), \
+            _create_compute(credentials)
 
     # Otherwise the credentials type must be credentials_token.
-    creds_from_token = AccessTokenCredentials(credentials_field, "MyAgent/1.0")
-    token_request = creds_from_token.authorize(httplib2.Http())
-    return _create_crm_oauth_token_request(token_request), \
-        _create_iam_oauth_token_request(token_request), \
-        _create_compute_oauth_token_request(token_request)
+    creds_from_token = Credentials(credentials_field)
```

Comment:
Can we dedup the `_create_*` logic by grouping the two "credential constructors" into a single if block. E.g.


```python
if cred_type == "service_account":
   credentials = service_account.Credentials.from_service_account_info(...)
elif cred_type == "token":
   credentials = oauth.Credentials(....)

return _create_crm(...), [...]
```

Reply:
That's a good idea, the current structure only makes sense when we had two different sets of _create_* functions